JacksonJsonHandler cannot parse task status enqueued

<!-- This is not an exhaustive model but a help. No step is mandatory. -->

**Description**
I would like to create a new index. The client uses a ``new JacksonJsonHandler()``. However, when reading the task state I get an error.

**Expected behavior**
Parsing string value ``enqueued`` as enum value ``ENQUEUED``.

**Current behavior**
Currently i get the following error:
```
java.lang.IllegalStateException: Failed to execute CommandLineRunner
	at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:773) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.SpringApplication.callRunners(SpringApplication.java:754) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:318) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1305) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1294) ~[spring-boot-3.1.0.jar:3.1.0]
	at de.skillkiller.clipindex.ClipIndexApplication.main(ClipIndexApplication.java:20) ~[classes/:na]
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:na]
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[na:na]
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:na]
	at java.base/java.lang.reflect.Method.invoke(Method.java:568) ~[na:na]
	at org.springframework.boot.devtools.restart.RestartLauncher.run(RestartLauncher.java:49) ~[spring-boot-devtools-3.1.0.jar:3.1.0]
Caused by: java.lang.RuntimeException: Meilisearch Exception: {com.fasterxml.jackson.databind.exc.InvalidFormatException. Error=com.fasterxml.jackson.databind.exc.InvalidFormatException: Cannot deserialize value of type `com.meilisearch.sdk.model.TaskStatus` from String "enqueued": not one of the values accepted for Enum class: [SUCCEEDED, PROCESSING, ENQUEUED, FAILED, CANCELED]
 at [Source: (String)"{"taskUid":1,"indexUid":"clips2","status":"enqueued","type":"indexCreation","enqueuedAt":"2023-05-20T00:20:14.811110336Z"}"; line: 1, column: 43] (through reference chain: com.meilisearch.sdk.model.TaskInfo["status"])}
	at com.meilisearch.sdk.http.response.BasicResponse.create(BasicResponse.java:22) ~[meilisearch-java-0.11.1.jar:na]
	at com.meilisearch.sdk.HttpClient.post(HttpClient.java:96) ~[meilisearch-java-0.11.1.jar:na]
	at com.meilisearch.sdk.IndexesHandler.createIndex(IndexesHandler.java:52) ~[meilisearch-java-0.11.1.jar:na]
	at com.meilisearch.sdk.Client.createIndex(Client.java:73) ~[meilisearch-java-0.11.1.jar:na]
	at de.skillkiller.clipindex.MeilisearchClient.lambda$checkIndexes$0(MeilisearchClient.java:48) ~[classes/:na]
	at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:770) ~[spring-boot-3.1.0.jar:3.1.0]
	... 10 common frames omitted
Caused by: com.meilisearch.sdk.exceptions.JsonDecodingException: com.fasterxml.jackson.databind.exc.InvalidFormatException: Cannot deserialize value of type `com.meilisearch.sdk.model.TaskStatus` from String "enqueued": not one of the values accepted for Enum class: [SUCCEEDED, PROCESSING, ENQUEUED, FAILED, CANCELED]
 at [Source: (String)"{"taskUid":1,"indexUid":"clips2","status":"enqueued","type":"indexCreation","enqueuedAt":"2023-05-20T00:20:14.811110336Z"}"; line: 1, column: 43] (through reference chain: com.meilisearch.sdk.model.TaskInfo["status"])
	at com.meilisearch.sdk.json.JacksonJsonHandler.decode(JacksonJsonHandler.java:72) ~[meilisearch-java-0.11.1.jar:na]
	at com.meilisearch.sdk.http.response.BasicResponse.create(BasicResponse.java:15) ~[meilisearch-java-0.11.1.jar:na]
	... 15 common frames omitted
Caused by: com.fasterxml.jackson.databind.exc.InvalidFormatException: Cannot deserialize value of type `com.meilisearch.sdk.model.TaskStatus` from String "enqueued": not one of the values accepted for Enum class: [SUCCEEDED, PROCESSING, ENQUEUED, FAILED, CANCELED]
 at [Source: (String)"{"taskUid":1,"indexUid":"clips2","status":"enqueued","type":"indexCreation","enqueuedAt":"2023-05-20T00:20:14.811110336Z"}"; line: 1, column: 43] (through reference chain: com.meilisearch.sdk.model.TaskInfo["status"])
	at com.fasterxml.jackson.databind.exc.InvalidFormatException.from(InvalidFormatException.java:67) ~[jackson-databind-2.15.0.jar:2.15.0]
	at com.fasterxml.jackson.databind.DeserializationContext.weirdStringException(DeserializationContext.java:2002) ~[jackson-databind-2.15.0.jar:2.15.0]
	at com.fasterxml.jackson.databind.DeserializationContext.handleWeirdStringValue(DeserializationContext.java:1230) ~[jackson-databind-2.15.0.jar:2.15.0]
	at com.fasterxml.jackson.databind.deser.std.EnumDeserializer._deserializeAltString(EnumDeserializer.java:415) ~[jackson-databind-2.15.0.jar:2.15.0]
	at com.fasterxml.jackson.databind.deser.std.EnumDeserializer._fromString(EnumDeserializer.java:279) ~[jackson-databind-2.15.0.jar:2.15.0]
	at com.fasterxml.jackson.databind.deser.std.EnumDeserializer.deserialize(EnumDeserializer.java:248) ~[jackson-databind-2.15.0.jar:2.15.0]
	at com.fasterxml.jackson.databind.deser.impl.FieldProperty.deserializeAndSet(FieldProperty.java:138) ~[jackson-databind-2.15.0.jar:2.15.0]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:314) ~[jackson-databind-2.15.0.jar:2.15.0]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:177) ~[jackson-databind-2.15.0.jar:2.15.0]
	at com.fasterxml.jackson.databind.deser.DefaultDeserializationContext.readRootValue(DefaultDeserializationContext.java:323) ~[jackson-databind-2.15.0.jar:2.15.0]
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4825) ~[jackson-databind-2.15.0.jar:2.15.0]
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3772) ~[jackson-databind-2.15.0.jar:2.15.0]
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3740) ~[jackson-databind-2.15.0.jar:2.15.0]
	at com.meilisearch.sdk.json.JacksonJsonHandler.decode(JacksonJsonHandler.java:65) ~[meilisearch-java-0.11.1.jar:na]
	... 16 common frames omitted

```

**Screenshots or Logs**
If applicable, add screenshots or logs to help explain your problem.

**Environment (please complete the following information):**
- OS: Windows 11
- Java: 17
- Meilisearch version: v.1.1.1
- meilisearch-java version: 0.11.1
- Spring Boot: 3.1.0

--- a/src/main/java/com/meilisearch/sdk/json/JacksonJsonHandler.java
+++ b/src/main/java/com/meilisearch/sdk/json/JacksonJsonHandler.java
@@ -9,7 +9,6 @@ import com.fasterxml.jackson.databind.util.StdDateFormat;
 import com.meilisearch.sdk.exceptions.JsonDecodingException;
 import com.meilisearch.sdk.exceptions.JsonEncodingException;
 import com.meilisearch.sdk.exceptions.MeilisearchException;
-import com.meilisearch.sdk.model.Key;
 import java.io.IOException;
 
 public class JacksonJsonHandler implements JsonHandler {
@@ -39,10 +38,7 @@ public class JacksonJsonHandler implements JsonHandler {
             return (String) o;
         }
         try {
-            // TODO: review later
-            if (o != null && o.getClass() != Key.class) {
-                this.mapper.setSerializationInclusion(Include.NON_NULL);
-            }
+            this.mapper.setSerializationInclusion(Include.NON_NULL);
             return mapper.writeValueAsString(o);
         } catch (JsonProcessingException e) {
             throw new JsonEncodingException(e);

--- a/src/main/java/com/meilisearch/sdk/model/Key.java
+++ b/src/main/java/com/meilisearch/sdk/model/Key.java
@@ -1,5 +1,7 @@
 package com.meilisearch.sdk.model;
 
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
 import java.util.Date;
 import lombok.Getter;
 import lombok.Setter;
@@ -11,6 +13,7 @@ import lombok.experimental.Accessors;
  * <p>https://www.meilisearch.com/docs/reference/api/keys
  */
 @Getter
+@JsonInclude(Include.NON_NULL)
 public class Key {
     @Setter
     @Accessors(chain = true)
@@ -34,6 +37,7 @@ public class Key {
 
     @Setter
     @Accessors(chain = true)
+    @JsonInclude(Include.ALWAYS)
     protected Date expiresAt = null;
 
     protected Date createdAt = null;

--- a/src/main/java/com/meilisearch/sdk/model/MatchingStrategy.java
+++ b/src/main/java/com/meilisearch/sdk/model/MatchingStrategy.java
@@ -1,7 +1,12 @@
 package com.meilisearch.sdk.model;
 
+import com.fasterxml.jackson.annotation.JsonValue;
+import com.google.gson.annotations.SerializedName;
+
 public enum MatchingStrategy {
+    @SerializedName("all")
     ALL("all"),
+    @SerializedName("last")
     LAST("last");
 
     public final String matchingStrategy;
@@ -10,6 +15,7 @@ public enum MatchingStrategy {
         this.matchingStrategy = matchingStrategy;
     }
 
+    @JsonValue
     @Override
     public String toString() {
         return this.matchingStrategy;

--- a/src/main/java/com/meilisearch/sdk/model/TaskStatus.java
+++ b/src/main/java/com/meilisearch/sdk/model/TaskStatus.java
@@ -1,5 +1,7 @@
 package com.meilisearch.sdk.model;
 
+import com.fasterxml.jackson.annotation.JsonValue;
+
 public enum TaskStatus {
     ENQUEUED("enqueued"),
     PROCESSING("processing"),
@@ -13,6 +15,7 @@ public enum TaskStatus {
         this.taskStatus = taskStatus;
     }
 
+    @JsonValue
     @Override
     public String toString() {
         return this.taskStatus;

--- a/src/main/java/com/meilisearch/sdk/model/TaskStatus.java
+++ b/src/main/java/com/meilisearch/sdk/model/TaskStatus.java
@@ -1,12 +1,18 @@
 package com.meilisearch.sdk.model;
 
 import com.fasterxml.jackson.annotation.JsonValue;
+import com.google.gson.annotations.SerializedName;
 
 public enum TaskStatus {
+    @SerializedName("enqueued")
     ENQUEUED("enqueued"),
+    @SerializedName("processing")
     PROCESSING("processing"),
+    @SerializedName("succeeded")
     SUCCEEDED("succeeded"),
+    @SerializedName("failed")
     FAILED("failed"),
+    @SerializedName("canceled")
     CANCELED("canceled");
 
     public final String taskStatus;

--- a/src/test/java/com/meilisearch/integration/TasksTest.java
+++ b/src/test/java/com/meilisearch/integration/TasksTest.java
@@ -21,6 +21,7 @@ public class TasksTest extends AbstractIT {
     @BeforeEach
     public void initialize() {
         this.setUp();
+        this.setUpJacksonClient();
         if (testData == null) testData = this.getTestData(MOVIES_INDEX, Movie.class);
     }
 
@@ -252,5 +253,17 @@ public class TasksTest extends AbstractIT {
         index.waitForTask(task.getTaskUid());
 
         assertThrows(Exception.class, () -> index.waitForTask(task.getTaskUid(), 0, 50));
+    }
+
+    /** Test Tasks with Jackson Json Handler */
+    @Test
+    public void testTasksWithJacksonJsonHandler() throws Exception {
+        String indexUid = "tasksWithJacksonJsonHandler";
+        Index index = clientJackson.index(indexUid);
+
+        TestData<Movie> testData = this.getTestData(MOVIES_INDEX, Movie.class);
+        TaskInfo task = index.addDocuments(testData.getRaw());
+
+        assertEquals(TaskStatus.ENQUEUED, task.getStatus());
     }
 }

--- a/src/test/java/com/meilisearch/integration/classes/AbstractIT.java
+++ b/src/test/java/com/meilisearch/integration/classes/AbstractIT.java
@@ -41,8 +41,7 @@ public abstract class AbstractIT {
     }
 
     public void setUpJacksonClient() {
-        Config config = new Config(getMeilisearchHost(), "masterKey");
-        config.setJsonHandler(new JacksonJsonHandler());
+        Config config = new Config(getMeilisearchHost(), "masterKey", new JacksonJsonHandler());
         if (clientJackson == null) clientJackson = new Client(config);
     }
 

